### PR TITLE
[usbdev] usbdev scoreboard PR

### PR DESCRIPTION
**Pull request by Saad22386**
The Purpose of this PR is to sync the scoreboard with usbdev env so slight changes are done in some classes based on requirement.

### CODE REVIEW

**Comment by Saad22386** [comment]
@jdonjdon why last two CI checks are failing.

**Comment by Saad22386** [comment]
@jdonjdon kindly review it.

File: hw/dv/sv/usb20_agent/usb20_monitor.sv [hunk 1]
```diff
@@ -86,13 +85,13 @@ class usb20_monitor extends dv_base_monitor #(
     sof_result = sof_p'(bit_destuffed);
 
     `uvm_info(`gfn, $sformatf("Final Sof Packet = %b", sof_result), UVM_DEBUG)
-    m_sof_pkt.m_pid_type = pid_type_e'(sof_result[9:2]);
+    m_sof_pkt.m_pid_type = sof_result[9:2];
```
Comment (jdonjdon) [inline hw/dv/sv/usb20_agent/usb20_monitor.sv:88]: why did you remove strong type casting?
This would be good to have because we have other partners using xeclium, which requires such strong type casting.
@rswarbrick, can you confirm?

Comment (Saad22386) [reply hw/dv/sv/usb20_agent/usb20_monitor.sv:88]: > why did you remove strong type casting? This would be good to have because we have other partners using xeclium, which requires such strong type casting. @rswarbrick, can you confirm?

this was not included in our last merged monitor file i haven't rebase it that's why we missed type casting but I'll add type casting.

Comment (alees24) [reply hw/dv/sv/usb20_agent/usb20_monitor.sv]: When casting from bits/logic to an enum type, an explicit cast is required with xcelium. vcs, sadly, defaults to not enforcing the 'strong typing' rules of SystemVerilog, which is how some of the code ended up trying to cast/compare between enum types and string types.
We do appreciate that you do not have access to both tools.

Comment (Saad22386) [reply hw/dv/sv/usb20_agent/usb20_monitor.sv]: @jdonjdon now i have added strong type casting.

**Review by jdonjdon — APPROVED** [review]
LGTM, thanks for fixing type casting issue.

**Comment by Saad22386** [comment]
> LGTM, thanks for fixing type casting issue.

Thanks @jdonjdon  for valuable feedback 
Look forward for other changes approval.
